docs/tutorial/creating_your_first_admin_class/defining_entities.rst

```diff
@@ -217,45 +217,91 @@ Comment
          * @ORM\Column(type="integer")
          * @ORM\GeneratedValue(strategy="AUTO")
          */
-        protected $id;
+        private $id;
 
         /**
          * @ORM\Column(type="string")
          *
          * @Assert\NotBlank()
          */
-        protected $name;
+        private $name;
 
 
         /**
          * @ORM\Column(type="string")
          *
          * @Assert\NotBlank()
          */
-        protected $email;
+        private $email;
 
         /**
          * @ORM\Column(type="string")
          */
-        protected $url;
+        private $url;
 
         /**
          * @ORM\Column(type="text")
          * @Assert\NotBlank()
          */
-        protected $message;
+        private $message;
 
         /**
          * @ORM\ManyToOne(targetEntity="Post")
          */
-        protected $post;
+        private $post;
 
         public function __toString()
         {
             return $this->getName();
         }
     }
 
+.. note::
+
+    For advanced usage, ``$id`` might be implemented as an object. The bundle will automatically resolve its string
+    representation from the ID object using ``$entity->getId()->__toString()`` (if implemented) when needed
+    (e.g., for generating url / rendering).
+
+    For example, in a use case where `InnoDB-optimised binary UUIDs`_ is implemented:
+
+    .. code-block:: php
+
+        <?php
+
+        class Comment
+        {
+            /**
+             * @var \Ramsey\Uuid\UuidInterface
+             * @Id
+             * @Column(type="uuid_binary_ordered_time", unique=true)
+             * @GeneratedValue(strategy="CUSTOM")
+             * @CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidOrderedTimeGenerator")
+             */
+            private $id;
+
+            // ...
+        }
+
+    For more details, ``\Ramsey\Uuid\Uuid`` implements ``\Ramsey\Uuid\UuidInterface`` and has offered ``__toString``
+    methods as shown in below pseudo code:
+
+    .. code-block:: php
+
+        <?php
+        namespace Ramsey\Uuid;
```

Comment:
this should really be something like `App\...` and just import the Ramsey uuid

Reply:
This part supposes to be offered by ramsey/uuid package in the example. Seems it is causing confusion as I show its implementation here. Should I remove this part instead?